### PR TITLE
[docs-revamp] - Clean up metadata guide

### PR DESCRIPTION
**Pull request by erinkcochran87**
## Summary & Motivation

This PR cleans up the Asset metadata guide in the new docs.

## Changelog

Insert changelog entry or "NOCHANGELOG" here.

- [ ] `NEW` _(added new feature or capability)_
- [ ] `BUGFIX` _(fixed a bug)_
- [ ] `DOCS` _(added or updated documentation)_

### CODE REVIEW

File: docs/docs-beta/docs/guides/metadata.md [hunk 1]
```diff
@@ -5,9 +5,14 @@ sidebar_position: 40
...
 
-Assets feature prominently in the Dagster UI. It's often helpful to attach information to assets to understand where they're stored, what they contain, and how they should be organized.
+Assets feature prominently in the Dagster UI. Attaching information to assets allows you to understand where they're stored, what they contain, and how they should be organized.
 
-In Dagster, you can attach ownership information, organize your assets with tags, attach rich, complex information with metadata, and link your assets with their source code.
+Using metadata in Dagster, you can:
+
+- Attach ownership information
+- Organize assets with tags
+- Attach rich, complex information
```
Comment (PedramNavid) [inline docs/docs-beta/docs/guides/metadata.md:14]: Rich, complex information is vague

File: docs/docs-beta/docs/guides/metadata.md [hunk 2]
```diff
@@ -5,9 +5,14 @@
...
+- Link assets with their source code
 
 ## What you'll learn
```
[the remark this answers is not in the document]
Comment (erinkcochran87) [reply docs/docs-beta/docs/guides/metadata.md:17]: They should've been; I'll take care of it

File: docs/docs-beta/docs/guides/metadata.md [hunk 3]
```diff
@@ -54,47 +55,37 @@ Here's an example of some tags you might apply to an asset:
...
-<CodeExample filePath="guides/data-modeling/metadata/tags.py" language="python" title="Using tags" />
+<CodeExample filePath="guides/data-modeling/metadata/tags.py" language="python" />
 
 Keep in mind that tags must contain only strings as keys and values. Additionally, the Dagster UI will render tags with the empty string as a "label" rather than a key-value pair.
 
-## Attaching metadata to an asset at definition time
+## Attaching metadata to assets \{#attaching-metadata}
 
-Attaching metadata at definition time is similar to how you attach tags.
+**Metadata** allows you to attach rich information to the asset, like a Markdown description, a table schema, or a time series. Metadata is more flexible than tags, as it can store more complex information. Metadata can be attached to an asset at definition time (that's, when the code is first imported) or at runtime (every time an asset is materialized).
```
[the remark this answers is not in the document]
Comment (erinkcochran87) [reply docs/docs-beta/docs/guides/metadata.md:66]: Good catch, ty

File: docs/docs-beta/docs/guides/metadata.md [hunk 4]
```diff
@@ -54,47 +55,37 @@ Here's an example of some tags you might apply to an asset:
...
+
+### At definition time \{#definition-time-metadata}
+
+Attaching metadata at definition time is similar to how tags are attached:
```
[the remark this answers is not in the document]
Comment (erinkcochran87) [reply docs/docs-beta/docs/guides/metadata.md:70]: Yeah, tbh I'm not sure this comparison lands even if you did read tags. I'll replace it with something else.

File: docs/docs-beta/docs/guides/metadata.md (hunk 1 above)
Comment (erinkcochran87) [reply docs/docs-beta/docs/guides/metadata.md:14]: An example would help here - maybe mentioning something about Markdown?

File: docs/docs-beta/docs/guides/metadata.md [hunk 5]
```diff
@@ -106,75 +97,71 @@ Some metadata keys will be given special treatment in the Dagster UI.
...
+## Linking assets with source code \{#source-code}
 
-This feature is considered experimental and is under active development. This guide will be updated as we roll out new features.
+import Experimental from '../partials/\_Experimental.md';
 
-:::
+<Experimental />
 
-Attaching code reference metadata to your Dagster asset definitions allows you to easily view those assets' source code from the Dagster UI both in local development and in production.
+Attaching code reference metadata to Dagster asset definitions allows you to easily view those assets' source code from the Dagster UI, both in local development and in production.
```
[the remark this answers is not in the document]
Comment (erinkcochran87) [reply docs/docs-beta/docs/guides/metadata.md:132]: > To link assets with their source code, you can attach a **code reference**. Code references are a type of metadata that allow you to easily view those assets' source code from the Dagster UI, both in local development and in production.

Thoughts?

File: docs/docs-beta/docs/guides/metadata.md [hunk 6]
```diff
@@ -54,47 +55,37 @@ Here's an example of some tags you might apply to an asset:
...
 
 To learn more about the different types of metadata you can attach, see the [`MetadataValue`](/todo) API docs.
 
 Some metadata keys will be given special treatment in the Dagster UI. See the [Standard metadata types](#standard-metadata-types) section for more information.
 
-## Attaching metadata to an asset at runtime
+### At runtime \{#runtime-metadata}
 
-Metadata becomes powerful when it's attached when an asset is materialized. This allows you to update metadata when information about an asset changes and track historical metadata such as execution time and row counts as a time series.
+Metadata attached to assets at materialization allows you to update an asset's information when it changes and track historical metadata, such as execution time and row counts, as a time series.
```
[the remark this answers is not in the document]
Comment (erinkcochran87) [reply docs/docs-beta/docs/guides/metadata.md]: Updated

**Review by cmpadden — APPROVED** [review]

**Pull request merged**